torch/_higher_order_ops/utils.py

```diff
@@ -135,7 +135,13 @@ def _has_potential_branch_input_alias(branch, inputs):
     bit restrictive as the branch must be traceable.
     """
     try:
-        gm = make_fx(branch)(*inputs)
+        # At this point, we could be invoking this when PreDispatch key is active. So we manually skip it
+        include_set = torch._C._dispatch_tls_local_include_set().remove(
+            torch._C.DispatchKey.PreDispatch
+        )
+        exclude_set = torch._C._dispatch_tls_local_exclude_set()
+        with torch._C._ForceDispatchKeyGuard(include_set, exclude_set):
+            gm = make_fx(branch)(*inputs)
```

Comment:
We don't need this if we thread `pre_dispatch` in (which we are doing), right?

Reply:
Yep!